### PR TITLE
subscription: Prevent Subscription spoke being mandatory with DVD source set (port to Fedora)

### PR DESCRIPTION
**Pull request by M4rtinK**
Looks like there is a chance for the Subscription spoke to be set mandatory if payload initialization takes too long and the Subscription spoke gets initialized before payload is done.

The first commit make the Subscription spoke initialization to wait on the payload thread to finish.

The second commit enhances this by moving the waiting to a separate thread, preventing it from potentially
blocking the GUI startup for a noticeable amount of time.

### CODE REVIEW

**Comment by M4rtinK** [comment]
Original RHEL PR: #2725

File: pyanaconda/ui/gui/spokes/subscription.py [hunk 1]
```diff
@@ -126,8 +127,16 @@ def __init__(self, *args):
 
     @property
     def ready(self):
-        """The subscription spoke is always ready."""
-        return True
+        """The subscription spoke is ready once its initialization thread finishes.
+
+        We do this to avoid the Subscription spoke being set mandatory in cases
+        where the current installation source is the CDN, but payload refres is still
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/subscription.py:133]:
```suggestion
        where the current installation source is the CDN, but payload refresh is still
```

**Review by jkonecny12 — APPROVED** [review]
Looks good to me otherwise.

File: pyanaconda/ui/gui/spokes/subscription.py (hunk 1 above)
Comment (M4rtinK) [reply pyanaconda/ui/gui/spokes/subscription.py:133]: Thanks, fixed! :)

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/ui/gui/spokes/subscription.py [hunk 2]
```diff
@@ -126,8 +127,16 @@ def __init__(self, *args):
...
     @property
     def ready(self):
-        """The subscription spoke is always ready."""
-        return True
+        """The subscription spoke is ready once its initialization thread finishes.
+
+        We do this to avoid the Subscription spoke being set mandatory in cases
+        where the current installation source is the CDN, but payload refresh is still
+        running and it might change to CDROM later one. We achieve this by waiting
+        for tha payload refresh thread to finish in the Subscription spoke initialization
```
Comment (rvykydal) [inline pyanaconda/ui/gui/spokes/subscription.py:135]: "for the"

File: pyanaconda/ui/gui/spokes/subscription.py [hunk 3]
```diff
@@ -634,8 +643,19 @@ def initialize(self):
...
+        # start the rest of spoke initialization which might take some time
+        # (mainly due to waiting for various initialization threads to finish)
+        # in a separate thread
+        threadMgr.add(AnacondaThread(name=THREAD_SUBSCRIPTION_SPOKE_INIT,
+                                     target=self._initialize))
+
+    def _initialize(self):
         # wait for subscription thread to finish (if any)
         threadMgr.wait(THREAD_SUBSCRIPTION)
+        # also wait for the payload thread, which migh still be processing
```
Comment (rvykydal) [inline pyanaconda/ui/gui/spokes/subscription.py:655]: mihg -> might

**Review by rvykydal — APPROVED** [review]
Looks good to me, I just found two typos.

**Pull request merged**